hr_holidays_accrual_advanced/wizards/hr_leave_allocation_accrual_calculator.py

```diff
@@ -60,17 +60,17 @@ class HrLeaveAllocationAccrualBalanceCalculator(models.TransientModel):
         'date',
     )
     def _onchange(self):
-        self._recalculate()
-
-    @api.multi
-    def _recalculate(self):
-        self.ensure_one()
+        CalculatorAccruement = self.env[
+            'hr.leave.allocation.accrual.calculator.accruement'
+        ]
+        accruement_ids = CalculatorAccruement
 
         if not self.date:
             self.write({
```

Comment:
OK, now that you have removed the intermediate method I see the problem: you can't do a write on an onchange. You must do `.update` in replacement of.